### PR TITLE
feat: Add additional utilities for real-time app inspection

### LINKED ISSUE #8787: Implement basic utilities for Monitoring tool
### Description

Related to: https://github.com/status-im/status-desktop/issues/8711

Basic utilities written in c++ which could be later easily used from qml to build Monitor app's UI.
For example:
- taking name of all context properties injected from the backend
- taking role names for given model
- taking object by their name

### Acceptance Criteria

C++ utils exposed to qml in a non-intrusive way for the main app in release mode.

### PR DESCRIPTION
**Pull request by micieslak**
## What does the PR do

Adds some utilities necessary to inspect properties injected from a backend.

- context properties names exposed as a model
- checking if object is a model
- exposing human-readable type name
- exposing role names for models

Closes: https://github.com/status-im/status-desktop/issues/8787

### CODE REVIEW

File: lib/include/DOtherSide/Status/Monitoring/Monitor.h [hunk 1]
```diff
@@ -9,21 +12,23 @@ class QJSEngine;
...
     Monitor() = default;
 
 public:
     void initialize(QQmlApplicationEngine *engine);
-    QStringList getContextPropertiesNames() const;
+    ContextPropertiesModel* contexPropertiesModel();
     void addContextPropertyName(const QString &contextPropertyName);
 
+    Q_INVOKABLE bool isModel(QVariant obj) const;
+    Q_INVOKABLE QString typeName(QVariant obj) const;
```
Comment (caybro) [inline lib/include/DOtherSide/Status/Monitoring/Monitor.h:26]: Pass `QVariant` as const ref (`const QVariant &obj`)

**Review by caybro — APPROVED** [review]
Looks good, just one minor thing

File: lib/src/Status/Monitoring/Monitor.cpp [hunk 2]
```diff
@@ -28,18 +28,59 @@ void Monitor::initialize(QQmlApplicationEngine* engine) {
...
+    if (engine == nullptr)
+        return {};
+
+    const auto& roleNames = model->roleNames();
+
+    QJSValue array = engine->newArray(roleNames.size());
+    QList<int> keys = roleNames.keys();
+
+    for (auto i = 0; i < keys.size(); i++) {
+        QJSValue item = engine->newObject();
```
Comment (caybro) [inline lib/src/Status/Monitoring/Monitor.cpp:73]: Oh and, careful about the ownership here:

```
QJSValue QJSEngine::newQObject(QObject *object) Creates a JavaScript object that wraps the given QObject object, using JavaScriptOwnership.
```

Comment (micieslak) [reply lib/src/Status/Monitoring/Monitor.cpp:73]: But do you see any problems here? I can't see any danger here, final object doesn't depend on the input object, it's independent.

**Review by osmaczko — APPROVED** [review]

**Pull request merged**